mozci/push.py

```diff
@@ -70,20 +74,35 @@ class Push:
 
     def __init__(self, revs, branch="autoland"):
         if isinstance(revs, str):
+            # Direct usage of a single revision reference
             self._revs = None
-            revs = [revs]
+            head_revision = revs
+            self._bugs = None
+        elif (
+            isinstance(revs, list)
+            and len(revs) > 0
+            and all(map(lambda r: isinstance(r, dict), revs))
+        ):
+            # We should get detailed revision objects here
+            # and get the list of Bugzilla Bug Ids from the description
+            self._bugs = set(
+                itertools.chain(*[parse_bugs(rev.get("desc", "")) for rev in revs])
+            )
+
+            self._revs = [r["node"] for r in revs]
+            head_revision = self._revs[0]
         else:
-            self._revs = revs
+            raise NotImplementedError(f"Cannot process revisions: {revs}")
```

Comment:
Why not keeping support for passing a list of str?

Reply:
Because you mentionned using only full payloads (or a single revision to kickstart the push).

Is there a use case for a list of strings now ?

Reply:
Just as an alternative to single revision, if an external user wants to instantiate a Push directly